pyteal/ast/abi/method_return.py

```diff
@@ -30,7 +29,7 @@ def __teal__(self, options: "CompileOptions") -> Tuple[TealBlock, TealSimpleBloc
         start, end = Log(Concat(Bytes(RETURN_HASH_PREFIX), self.arg.encode())).__teal__(
             options
         )
-        NatalStackFrame.reframe_ops_in_blocks(self._sframes_container or self, start)
+        NatalStackFrame.reframe_ops_in_blocks(self, start)
```

Comment:
qq: could you explain this line, why `_sframes_container` is removed in `MethodReturn`?

I suppose the side effect is that, the `root_expr` of `start` (and all following blocks of `start`) is tied to `self`, which is the instance of `MethodReturn`.

Reply:
At an earlier stage of this work, I used a more "cautious" approach that instead of overwriting "useless" `stack_frames` belonging to an `Expr` with a useful instance -as is happening now- I tried to pass the information around in associated `TealBlocks` and also keep the original `expr` and `stack_frames`.   I found this indirect approach to be harder to manage then simply overwriting. I still use this indirect approach in conditional expressions, and `SubRoutine`s but I found that it was basically unworkable for the `Router`.